src/npm/Prefetcher.ts

```diff
@@ -35,31 +46,22 @@ export class Prefetcher {
     return this.#offset + this.#limit - this.#ready.length;
   }
 
-  get idleCount(): number {
-    return this.#ready.length;
-  }
-
   get isFinished(): boolean {
     return this.#finished;
   }
 
-  async getNext(): Promise<PrefetchedPkg> {
-    while (this.#ready.length <= 0) {
-      await wait(100);
-    }
+  run(): void {
+    this.#running = true;
 
-    return this.#ready.shift()!;
+    this.runInternal().catch((e) => {
+      sentry.report(e);
+    });
   }
 
-  async launch(): Promise<void> {
-    this.#running = true;
+  async runInternal(): Promise<void> {
     while (this.#running) {
-      if (this.#ready.length >= this.#maxIdle) {
-        await wait(config.prefetchWaitBetweenPage);
-        continue;
-      }
-
       await this.fetchOnePage();
+      await wait(ms('1 second'));
```

Comment:
Might be clearer to rename `fetchOnePage` to something more accurate now this it doesn't just fetch